rust_src/src/textprop.rs

```diff
@@ -28,7 +29,39 @@ pub fn get_char_property(position: EmacsInt, prop: LispObject, object: LispObjec
 /// If POSITION is at the end of OBJECT, the value is nil.
 #[lisp_fn(min = "2")]
 pub fn get_text_property(position: EmacsInt, prop: LispObject, object: LispObject) -> LispObject {
-    unsafe { textget(Ftext_properties_at(position.into(), object), prop) }
+    unsafe { textget(text_properties_at(position.into(), object), prop) }
+}
+
+/// Return the list of properties of the character at POSITION in OBJECT.
+/// If the optional second argument OBJECT is a buffer (or nil, which means
+/// the current buffer), POSITION is a buffer position (integer or marker).
+/// If OBJECT is a string, POSITION is a 0-based index into it.
+/// If POSITION is at the end of OBJECT, the value is nil.
+#[lisp_fn(min = "1")]
+pub fn text_properties_at(mut position: LispObject, mut object: LispObject) -> LispObject {
+    if object.is_nil() {
+        object = ThreadState::current_buffer().into();
+    }
+
+    let position_pointer = &mut position as *mut LispObject;
+    let i: *mut Lisp_Interval =
+        unsafe { validate_interval_range(object, position_pointer, position_pointer, false) };
+
+    if ptr::eq(i, ptr::null_mut()) {
```

Comment:
```suggestion
    if i.is_null() {
```

`ptr` provides `is_null`.